orchestration/transfer_controller.py

```diff
@@ -0,0 +1,223 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from dotenv import load_dotenv
+from enum import Enum
+import logging
+import os
+import time
+from typing import Generic, Protocol, TypeVar
+
+import globus_sdk
+
+from orchestration.flows.bl832.config import Config832
+from orchestration.globus.transfer import GlobusEndpoint, start_transfer
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+load_dotenv()
+
+
+Endpoint = TypeVar("Endpoint", bound=GlobusEndpoint)
+
+
+class BaseEndpoint(Protocol):
+    """
+    A protocol or abstract interface that all endpoints must implement or satisfy.
+    """
+
+    @property
+    @abstractmethod
+    def root_path(self) -> str:
+        """
+        Root path or base directory for this endpoint.
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """
+        A human-readable or reference name for the endpoint.
+        """
+        ...
+
+
+@dataclass
+class FileSystemEndpoint(BaseEndpoint):
+    root_path: str
+    name: str = "local"
+
+    def full_path(self, path_suffix: str) -> str:
+        if path_suffix.startswith("/"):
+            path_suffix = path_suffix[1:]
+        return f"{self.root_path.rstrip('/')}/{path_suffix}"
```

Comment:
You're using root_path, but it's never set anywhere. It's defined as a property in the super class but there's no setter defined there.